tests/unit/common/media_test.cpp

```diff
@@ -0,0 +1,329 @@
+/// media tests
+
+#include <string.h>
+#include <iostream>
+#include <cstdint>
+#include <vector>
+#include <deque>
+
+//#define CATCH_CONFIG_MAIN // This tells Catch to provide a main() - only do this in one cpp file
+#include "catch2/catch.hpp"
+
+#include "gcode_input_filter.hpp"
+
+using Catch::Matchers::Equals;
+
+struct GCRec {
+    std::string line;
+    uint32_t lastSuccessfulFileOffset;
+    GCRec(const char *gc, uint32_t o)
+        : line(gc)
+        , lastSuccessfulFileOffset(o) {}
+};
+
+using GCodesDeq = std::deque<GCRec>;
+
+// synthetic tests on memory buffers
+static const char gcodeTooLong[] = "G1\n"
+                                   "G2 ; with comment\n"
+                                   "; G3 commented completely\n"
+                                   "G4 with deliberately insane string behind it to exceed the 96 bytes of input buffer 12345678901234567890\n"
+                                   "; if(condition slicer profile){ G1 G2 G3 G4 G5 G6 G7 G8 G9 G10 }else{ G1 G2 G3 G4 G5 G6 G7 G8 G9 G10 } G1 G2 G3 G4 G5 G6 G7 G8 G9 G10 G1 G2 G3 G4 G5 G6 G7 G8 G9 G10 \n";
+
+static const char gcode0[] = "G1\n"
+                             "G2 ; with comment\n"
+                             "; G3 commented completely\n"
+                             "G4 with normal string behind within 96 bytes of input buffer\n"
+                             "; if(condition slicer profile){ G1 G2 G3 G4 G5 G6 G7 G8 G9 G10 }else{ G1 G2 G3 G4 G5 G6 G7 G8 G9 G10 } G1 G2 G3 G4 G5 G6 G7 G8 G9 G10 G1 G2 G3 G4 G5 G6 G7 G8 G9 G10 \n";
+
+constexpr uint32_t gcBuffSize = 97;
+
+enum EMediaLoop { OK,
+    ERROR,
+    ONHOLD };
+
+template <
+    typename ENQUEUEGCODE,
+    uint32_t RDBUFSIZE = 64,
+    uint32_t GCBUFSIZE = gcBuffSize>
+EMediaLoop media_loop_inner(ENQUEUEGCODE enq) {
```

Comment:
Please try to cover the edge cases as much as possible by emplace the invalid characters into the defined strings. It allows us to produce more robust unit tests only for the logic of the filter as we need.